### DISCUSSION TITLE
Grid layout with specific width and height for some cells

### DISCUSSION THREAD

**Discussion by janemba** [Textualize/textual#4838] [Q&A] [answered]
Hello,

I'm playing a bit with `textual` examples and I designed a layout that is convenient for me but I don't how to set the width and the height for specific cell in the grid.

Here is the code:
```
from textual.app        import App, ComposeResult
from textual.widgets import Static

class GridLayoutExample(App):
    CSS_PATH = "style.tcss"

    def compose(self) -> ComposeResult:

        yield Static("One", classes="box", id="one")
        yield Static("Two", classes="box", id="two")
        yield Static("Three", classes="box", id="three")
        yield Static("Four", classes="box", id="four")
        yield Static("Five", classes="box", id="five")

if __name__ == "__main__":
    app = GridLayoutExample()
    app.run()
```

Below the `CSS`:
```
Screen {
    layout: grid;
    grid-size: 2;
}

#one {
    column-span: 2;
}

#two {
    row-span: 2;
}

#three {
}

#four {
}

#five {
    column-span: 3;
}

.box {
    height: 100%;
    border: solid green;
}
```

The idea is to increase the width of the cell "two" and decrease the height of the cell "one" and "five". 

I tried to play with `grid-columns` for the cell two width but without success. How can it be performed ?

**Comment by willmcgugan** [comment]
You set `grid-columns` or `grid-rows` on the container, i.e. the screen in this case. Does that help?

**Comment by janemba** [comment]
@willmcgugan 
Unfortunately it doesn't help because it sets either for all columns or all rows.

The picture below is what I currently get. Lets say as an example I want the cell "Two" to be larger and the cell "Three" and "Four" with reduce width.
![Screenshot_2024-08-05_01 22 22](https://github.com/user-attachments/assets/af0a6cc1-4df2-4f26-8484-8b8541958806)

**Reply by willmcgugan** [reply]
I'm not clear what you are looking for. Suggest you sketch it.

**Reply by willmcgugan** [reply]
This?
<img width="1002" alt="Screenshot 2024-08-04 at 22 31 46" src="https://github.com/user-attachments/assets/f4a2e78c-b353-4d75-90a5-ad2a59385022">

**Reply by janemba** [reply]
I was doing the sketch but yes.

![sketch](https://github.com/user-attachments/assets/ceca7eb1-6c3d-4192-bb4d-e4711c95bc69)

**Reply by willmcgugan** [reply, marked as answer]
Add these rules to your Screen

```
grid-rows: 3 1fr 1fr 8;
grid-columns: 1fr 14;
```

**Reply by janemba** [reply]
Perfect !
